src/sdltiles.cpp

```diff
@@ -3070,6 +3090,44 @@ static void CheckMessages()
                         break;
                 }
                 break;
+#if defined(__IPHONEOS__)
+            case SDL_APP_WILLENTERBACKGROUND:
+                if( world_generator &&
+                    world_generator->active_world &&
+                    g && g->uquit == QUIT_NO &&
+                    !std::uncaught_exception() ) {
+                    g->quicksave();
+                }
+                if( SDL_IsTextInputActive() ) {
+                    // TODO: Abstract common method with above usage
+                    text_input_active_when_regaining_focus = true;
+                    // Stop text input to not intefere with other programs
+                    SDL_StopTextInput();
+                    // Clear uncommited IME text. TODO: commit IME text instead.
+                    last_input = input_event();
+                    last_input.type = input_event_t::keyboard_char;
+                    last_input.edit.clear();
+                    last_input.edit_refresh = true;
+                    text_refresh = true;
+                } else {
+                    text_input_active_when_regaining_focus = false;
+                }
+                break;
+            case SDL_APP_DIDENTERBACKGROUND:
+                window_focus = false;
+                break;
+            case SDL_APP_DIDENTERFOREGROUND:
+                /*if( !SDL_IsTextInputActive() ) {
+                    StopTextInput();
+                    StartTextInput();
+                } */
```

Comment:
Commented out code doesn't seem like a good idea, if it doesn't do anything we don't need it.

Reply:
Removed